test/manipulation.test.js

```diff
@@ -1097,6 +1098,71 @@ describe('manipulation', function() {
     });
   }
 
+  var hasReplaceById = !!getSchema().connector.replaceById;
+  bdd.describeIf(hasReplaceById, 'replaceOrCreate when forceId is true', function() {
+    var Post, foundId;
+    var ds = getSchema();
+    before(function(done) {
+      Post = ds.define('Post', {
+        title: {type: String, length: 255, index: true},
+        content: {type: String},
+        comments: [String],
+      }, {forceId: true});
+      ds.automigrate('Post', done);
+    });
+
+    it('fails when id does not exist in db', function(done) {
+      var post = {id: 123, title: 'a', content: 'AAA'};
+      Post.replaceOrCreate(post, function(err, p) {
+        err.statusCode.should.equal(404);
+        done();
+      });
+    });
+
+    it('works on create if the request does not include an ID', function(done) {
+      var post = {title: 'a', content: 'AAA'};
+      Post.replaceOrCreate(post, function(err, p) {
+        if (err) return done(err);
+        p.should.not.have.property('_id');
+        p.title.should.equal(post.title);
+        p.content.should.equal(post.content);
+        Post.findById(p.id, function(err, p) {
+          if (err) return done(err);
+          foundId = p.id;
+          done();
+        });
+      });
+    });
+
+    it('works on update if the request includes an exisiting ID in db', function(done) {
+      var post = {id: foundId, title: 'b', content: 'BBBB'};
+      Post.replaceOrCreate(post, function(err, p) {
+        if (err) return done(err);
+        p.id.should.equal(post.id);
+        p.should.not.have.property('_id');
+        p.title.should.equal(post.title);
+        p.content.should.equal(post.content);
+        done();
+      });
+    });
+
+    it('works without options on update', function(done) {
```

Comment:
I might miss the context of this test case, it looks same as "works on update if the request includes an exisiting ID in db", could you elaborate more which "option on update" is it checking?